### PR TITLE
New D210 check for white spaces.

### PR DESCRIPTION
**Pull request by jirikuncar**
This is a proposal for checking whitespaces in docstrings.

- [x] number
- [x] error message

### CODE REVIEW

**Comment by Nurdok** [comment]
First of all, thank you so much for investing in this. It's greatly appreciated. 

I'm currently on my phone, so I only skimmed the changes, but I have a few issues:
* Why did you choose the D290 error code? Always use the next available one in the appropriate category. 
* The check name should be more specific. The entire D2xx category is about whitespace issues, so it's not enough to call it that.

**Comment by Nurdok** [comment]
You should also update the README file with the new error.

**Comment by jirikuncar** [comment]
>> * [ ] number

> Why did you choose the D290 error code? Always use the next available one in the appropriate category.

I'm not sure about the number and error message. Do you have any proposal?

**Comment by jirikuncar** [comment]
@Nurdok based on `README`, I've changed the error code to 210. Is it ok?

**Comment by Nurdok** [comment]
D210 is okay, but please also address my other comments.

**Comment by jirikuncar** [comment]
Can you please provide the message?

File: pep257.py [hunk 1]
```diff
@@ -778,6 +778,17 @@ def check_newline_after_last_paragraph(self, definition, docstring):
                                  'quotes on separate line')
 
     @check_for(Definition)
+    def check_whitespaces(self, definition, docstring):
+        """D210: White spaces around doctring should be trimmed."""
+        if docstring:
+            lines = eval(docstring).split('\n')
+            if lines[0].startswith(' ') or \
+                    len(lines) == 1 and lines[0].endswith(' '):
+                return Error(
```
Comment (Nurdok) [inline pep257.py]: Put the line break in the string.

    return Error('D210: White spaces around docstring should '
                 'be trimmed.')

File: test_pep257.py [hunk 2]
```diff
@@ -76,7 +76,8 @@ def test_ignore_list():
     no blank line after one-liner is bad. Also this - \"\"\"
     return foo
     """
-    expected_error_codes = set(('D100', 'D400', 'D401', 'D205', 'D209'))
+    expected_error_codes = set((
+        'D100', 'D400', 'D401', 'D205', 'D209', 'D210'))
```
Comment (Nurdok) [inline test_pep257.py]: Break the line as part of the list:

    expected_error_codes = set(('D100', 'D400', 'D401', 'D205', 'D209',
                                'D210'))

File: pep257.py [hunk 3]
```diff
@@ -778,6 +778,17 @@ def check_newline_after_last_paragraph(self, definition, docstring):
                                  'quotes on separate line')
 
     @check_for(Definition)
+    def check_whitespaces(self, definition, docstring):
```
Comment (Nurdok) [inline pep257.py]: Give the check a more specific name, like `check_surrounding_whitespaces`.

File: pep257.py [hunk 4]
```diff
@@ -778,6 +778,17 @@ def check_newline_after_last_paragraph(self, definition, docstring):
...
+        """D210: White spaces around doctring should be trimmed."""
```
Comment (Nurdok) [inline pep257.py]: Change the message to `"D210: No whitespaces allowed surrounding docstring text."`

**Comment by jirikuncar** [comment]
@Nurdok thanks for your comments and time. :shipit:

File: pep257.py [hunk 5]
```diff
@@ -778,6 +778,16 @@ def check_newline_after_last_paragraph(self, definition, docstring):
                                  'quotes on separate line')
 
     @check_for(Definition)
+    def check_surrounding_whitespaces(self, definition, docstring):
+        """D210: White spaces around doctring should be trimmed."""
```
Comment (Nurdok) [inline pep257.py]: You should update the error message here as well (you updated only the README) and in the tests.

**Comment by Nurdok** [comment]
I left one more comment in the actual code - but notice that the build is now failing on your branch. Make sure your tests pass.

File: pep257.py (hunk 5 above)
Comment (jirikuncar) [inline pep257.py]: Ups. Sorry about it :(

**Comment by jirikuncar** [comment]
@Nurdok please note that I've also added `sudo: false` to `.travis.yml` in order to use container based infrastructure (see http://docs.travis-ci.com/user/workers/container-based-infrastructure/).

File: pep257.py [hunk 6]
```diff
@@ -778,6 +778,16 @@ def check_newline_after_last_paragraph(self, definition, docstring):
                                  'quotes on separate line')
 
     @check_for(Definition)
+    def check_surrounding_whitespaces(self, definition, docstring):
+        """D210: No whitespaces allowed surrounding docstring text."""
+        if docstring:
+            lines = eval(docstring).split('\n')
+            if lines[0].startswith(' ') or \
+                    len(lines) == 1 and lines[0].endswith(' '):
+                        return Error("D210: No whitespaces allowed "
```
Comment (Nurdok) [inline pep257.py]: This indentation looks wrong.

Comment (jirikuncar) [inline pep257.py]: Aaaah, broken `:autoindent` :(

**Comment by jirikuncar** [comment]
@Nurdok sorry for so many iterations. Please let me know when you release it on PyPI so I can remove it from my package https://github.com/jirikuncar/invenio-kwalitee/blob/master/invenio_kwalitee/kwalitee.py#L82-L93. Thanks

**Comment by Nurdok** [comment]
@jirikuncar Merged. It will probably take a few weeks before I make another release. In any case, thank you **so much** for investing the time and effort into this. I really can't thank you enough!

**Pull request merged**